kratos/utilities/enrichment_utilities.h

```diff
@@ -1973,6 +1973,599 @@ class EnrichmentUtilities
     }
     
     
+	//2D: 2 enrichment functions for capturing weak and strong discontinities. All the shape functions follow the criteria of Partition of Unity.  
+	static int CalculateEnrichedShapeFuncionsExtendedModified2D(Geometry< Node<3> >& Julio,boost::numeric::ublas::bounded_matrix<double,(2+1), 2 >& rPoints, boost::numeric::ublas::bounded_matrix<double, (2+1), 2 >& DN_DX,array_1d<double,(2+1)>& rDistances, array_1d<double,(3*(2-1))>& rVolumes, boost::numeric::ublas::bounded_matrix<double, 3*(2-1), (2+1) >& rGPShapeFunctionValues, array_1d<double,(3*(2-1))>& rPartitionsSign, std::vector<Matrix>& rGradientsValue, boost::numeric::ublas::bounded_matrix<double,3*(2-1), (5)>& NEnriched,boost::numeric::ublas::bounded_matrix<double,10, 2>& rGradientpositive,boost::numeric::ublas::bounded_matrix<double,10, 2>& rGradientnegative ,boost::numeric::ublas::bounded_matrix<int,3,3>& father_nodes,std::vector<Matrix>& PRUEBA, array_1d<double,6>& weight)
```

Comment:
Thanks for the comments. This function could be used to enrich temperature or velocity. For this reason, an example can be incorporated in the convection-diffusion application.